### DISCUSSION TITLE
Improvements to editing files on iPad

### DISCUSSION THREAD

**Discussion by anvme** [community/community#46395] [Mobile]
### Select Topic Area

Product Feedback

### Body

The edit option is awesome, BUT it needs some improvements:
1) Allow changing edit files window size, make it width much more
2) Option to edit files in full-screen mode, like the view files without a sidebar
3) Enable/Disable syntax highlight on edit

Because editing text on iPad Pro with /2 of full-screen size is sad.

**Comment by chris-c-thomas** [comment]
Are you using the iPadOS app or through the browser?

There are certainly code editors on iPadOS that'll give you better options but my favorite of late has been https://codesandbox.io/ios which you can sync your repositories and do some damn good code editing.

**Comment by anvme** [comment]
I need a Pro plan for Private repos.
GitHub is better, but edit window sucks

**Reply by Ratnomd** [reply]
swaffordray123@g- [ ] mail.com

**Reply by Ratnomd** [reply]
swaffordray123@gmail.com

**Reply by Ratnomd** [reply]
swaffordray123@gmail.com

**Reply by Nlhhh** [reply]
> [swaffordray123@gmail.com](mailto:swaffordray123@gmail.com)

Hii

**Comment by chris-c-thomas** [comment]
@anvme ahh I gotcha. I definitely still prefer GitHub in the browser on the iPad for this reason but the app for my iPhone. The app is getting better really quick so that's a good sign.

I use the beta version in TestFlight so I'll send in feedback about this on both my iPhone and iPad.

**Reply by anvme** [reply]
> @anvme ahh I gotcha. I definitely still prefer GitHub in the browser on the iPad for this reason but the app for my iPhone. The app is getting better really quick so that's a good sign.
> 
> I use the beta version in TestFlight so I'll send in feedback about this on both my iPhone and iPad.

Awesome!
Thank you

**Comment by alvarado18** [comment]
Hey good

**Comment by anvme** [comment]
@chris-c-thomas I found a better solution
1 simple function, allow editing file on the file view. (Additional check). As windows shows, it allows editing as in many code editors. With syntax highlight. That's it

**Comment by chris-c-thomas** [comment]
Well you can use **github.dev** from the file view. You can read about it [here](https://docs.github.com/en/codespaces/the-githubdev-web-based-editor). It's a VS Code editor in the web but not a full on environment like Codespaces is. And it's free. Here's some screenshots from my iPad. Use GitHub in the browser, not the app, and when you go to a file click the dropdown and you'll see `github.dev` in the top right.

![IMG_0001](https://user-images.githubusercontent.com/22353987/217116703-17782186-b192-41b3-97c3-8cb8b02de4fc.PNG)

---

![IMG_0002](https://user-images.githubusercontent.com/22353987/217116710-e9991029-6342-4266-bdb3-5fd7f2a830ff.PNG)

---

![IMG_0003](https://user-images.githubusercontent.com/22353987/217116718-5147af52-858f-437a-b3ff-7cbe9ed651ad.PNG)

This is all in Firefox on my iPad. It's the `https://github.com/github/docs` repo and the whole thing is opened in the editor to be worked on. Again, not a full dev environment like Codespaces so if you need compute, build tools, etc this might not be for you. But for editing it works well.
